go/engine/identify2_with_uid.go

```diff
@@ -148,7 +148,7 @@ func (e *Identify2WithUID) runReturnError(ctx *Context) (err error) {
 		return err
 	}
 
-	if !e.useAnyAssertions() && e.checkFastCacheHit() && e.allowEarlyOuts() {
+	if !e.useAnyAssertions() && e.allowEarlyOuts() && e.checkFastCacheHit() {
```

Comment:
It was an attempt to make things faster...no need to check the cache if no allowEarlyOuts, which is just a comparison.

But I'm sure negligible performance gain.